plugins/modules/idrac_license.py

```diff
@@ -0,0 +1,824 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+#
+# Dell OpenManage Ansible Modules
+# Version 8.7.0
+# Copyright (C) 2024 Dell Inc. or its subsidiaries. All Rights Reserved.
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+#
+
+
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+---
+module: idrac_license
+short_description: Configure iDRAC licenses
+version_added: "8.7.0"
+description:
+  - This module allows to import, export and delete licenses on iDRAC.
+extends_documentation_fragment:
+  - dellemc.openmanage.idrac_auth_options
+options:
+  license_id:
+    description:
+      - Entitlement ID of the license that is to be imported, exported or deleted.
+      - I(license_id) is required when I(delete) is C(true) or I(export) is C(true).
+    type: str
+    aliases: ['entitlement_id']
+  delete:
+    description:
+      - Delete the license from the iDRAC.
+      - When I(delete) is C(true), then I(license_id) is required.
+      - I(delete) is mutually exclusive with I(export) and I(import).
+    type: bool
+    default: false
+  export:
+    description:
+      - Export the license from the iDRAC.
+      - When I(export) is C(true), I(license_id) and I(share_parameters) is required.
+      - I(export) is mutually exclusive with I(delete) and I(import).
+    type: bool
+    default: false
+  import:
+    description:
+      - Import the license from the iDRAC.
+      - When I(import) is C(true), I(share_parameters) is required.
+      - I(import) is mutually exclusive with I(delete) and I(export).
+    type: bool
+    default: false
+  share_parameters:
+    description:
+      - Parameters that are required for the import and export operation of a license.
+      - I(share_parameters) is required when I(export) or I(import) is C(true).
+    type: dict
+    suboptions:
+      share_type:
+        description:
+          - Share type of the network share.
+          - C(local) uses local path for I(import) and I(export) operation.
+          - C(nfs) uses NFS share for I(import) and I(export) operation.
+          - C(cifs) uses CIFS share for I(import) and I(export) operation.
+          - C(http) uses HTTP share for I(import) and I(export) operation.
+          - C(https) uses HTTPS share for I(import) and I(export) operation.
+        type: str
+        choices: [local, nfs, cifs, http, https]
+        default: local
+      file_name:
```

Comment:
Shouldn't we specify supported file formats for import operation in the description?

Reply:
Done